test/Datadog.Trace.ClrProfiler.Native.Tests/IntegrationLoaderTests.cpp

```diff
@@ -0,0 +1,70 @@
+#include "pch.h"
+
+#include <sstream>
+
+#include "../../src/Datadog.Trace.ClrProfiler.Native/IntegrationLoader.h"
+
+TEST(IntegrationLoaderTest, HandlesMissingFile)
+{
+    auto integrations = IntegrationLoader::load_integrations_from_file(L"missing-file");
+    EXPECT_EQ(0, integrations.size());
+}
+
+TEST(IntegrationLoaderTest, HandlesInvalidIntegration)
+{
+    std::stringstream str("[{}]");
+    auto integrations = IntegrationLoader::load_integrations_from_stream(str);
+    // 0 because name is required
+    EXPECT_EQ(0, integrations.size());
+}
+
+TEST(IntegrationLoaderTest, HandlesSingleIntegrationWithNoMethods)
+{
+    std::stringstream str(R"TEXT(
+        [{ "name": "test-integration" }]
+    )TEXT");
+    auto integrations = IntegrationLoader::load_integrations_from_stream(str);
+    EXPECT_EQ(1, integrations.size());
+    EXPECT_STREQ(L"test-integration", integrations[0].integration_name.c_str());
+    EXPECT_EQ(0, integrations[0].method_replacements.size());
+}
+
+TEST(IntegrationLoaderTest, HandlesSingleIntegrationWithInvalidMethodReplacementType)
+{
+    std::stringstream str(R"TEXT(
+        [{ "name": "test-integration", "method_replacements": 1234 }]
+    )TEXT");
+    auto integrations = IntegrationLoader::load_integrations_from_stream(str);
```

Comment:
Add test where method_replacement is an object, but it's missing one of the keys (caller, target, wrapper), and one where the key is invalid (like a value type or something).